uvicorn/protocols/http/httptools_impl.py

```diff
@@ -414,11 +414,13 @@ def __init__(
         self.expected_content_length = 0
 
     # ASGI exception wrapper
-    async def run_asgi(self, app: "ASGI3Application") -> None:
+    async def run_asgi(self, app: ASGI3Application) -> None:
         try:
             result = await app(  # type: ignore[func-returns-value]
                 self.scope, self.receive, self.send
             )
+        except ClientDisconnected:
+            ...
```

Comment:
Do I need to call the `self.transport.close()`?

Reply:
Well... The `connection_lost` callback is called, and the transport is closed there.

Reply:
Actually, not adding the `self.transport.close()` means that we maintain the same behavior as before (which is better).